mrp_subcontracting_skip_no_negative/models/stock_move.py

```diff
@@ -14,11 +14,21 @@ def _action_done(self, cancel_backorder=False):
         # For rather unlikely occassions where linked production is not in the right
         # state.
         for move in moves_with_no_check:
-            production_move = self.search([("move_dest_ids", "=", move.id)])
-            production = production_move.production_id
-            if production.reservation_state != "assigned":
+            production_move = self.search(
+                [
+                    ("move_dest_ids", "=", move.id),
+                    ("state", "not in", ("done", "cancel")),
+                ]
+            )
+            productions = production_move.production_id
+            unassigned_productions = productions.filtered(
+                lambda p: p.reservation_state != "assigned"
+            )
+            for production in unassigned_productions:
```

Comment:
```suggestion
            unassigned_productions.action_assign()
```
and remove the following line.